### PR TITLE
Add biggest bust and surprise to SeasonStatistics 

### PR DESCRIPTION
**Pull request by lrs8810**

### CODE REVIEW

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -1,15 +1,108 @@
...
+    @games.each do |game_id, game|
+      if game.season == season_id
+        filtered_games.push(game)
+      end
     end
-    games_by_team.each do |team_id, games|
-      ps_game = games.find_all do |game|
-        @games[game.game_id].type == "Postseason"
+
+    team_ids = @game_teams.map {|team| team.team_id}.uniq
```
Comment (grwthomps) [inline lib/season_statistics.rb:12]: Nice use of uniq!

File: lib/season_statistics.rb [hunk 2]
```diff
@@ -1,15 +1,108 @@
...
+          end
+        end
+        biggest_bust[team_id][season_type][:win_avg] = win_avg
+      end
+    end
+
+    team_with_biggest_bust = nil
+    biggest_bust_difference = 0
+    biggest_bust.each do |team_id, seasons|
+      seasons[:regular_season][:win_avg] = 0 if seasons[:regular_season][:win_avg].nan?
```
Comment (grwthomps) [inline lib/season_statistics.rb:45]: Nice solution for eliminating `NaN`!

File: lib/season_statistics.rb [hunk 3]
```diff
@@ -1,15 +1,108 @@
...
 
-  def biggest_bust(season)
-    games_by_team = @game_teams.group_by do |game_team|
-      game_team.team_id
+  def biggest_bust(season_id)
+    filtered_games = []
+    @games.each do |game_id, game|
+      if game.season == season_id
+        filtered_games.push(game)
+      end
```
Comment (grwthomps) [inline lib/season_statistics.rb:9]: Once we've created a helper method for `filtered_games`, we'll be able to shrink this.

**Review by hillstew — APPROVED** [review]

File: lib/season_statistics.rb [hunk 4]
```diff
@@ -1,15 +1,108 @@
...
+      difference = seasons[:regular_season][:win_avg] - seasons[:postseason][:win_avg]
+      if difference > biggest_bust_difference
+        biggest_bust_difference = difference
+        team_with_biggest_bust = team_id
+      end
+    end
+    @teams[team_with_biggest_bust].team_name
+  end
+
+  def biggest_surprise(season_id)
```
Comment (hillstew) [inline lib/season_statistics.rb:56]: Glad we were able to work through both of these this morning!

**Review by tyladevon — COMMENTED** [review]
Great resolution to the NaN issue in Biggest_bust method. Will complete Rival during work time today.

**Pull request merged**